web/modules/weather_data/src/Service/WeatherDataService.php

```diff
@@ -370,31 +360,33 @@ public function getApiObservationKey($observation)
          */
         $icon = $observation->icon;
 
-        if ($icon == null or strlen($icon) == 0) {
-            return "no data";
-        }
-
-        $url = parse_url($observation->icon);
-        $path = $url["path"];
-        $path = explode("/", $path);
-
-        // An icon url, when split to path parts,
-        // with have either 5 or 6 parts.
-        // Thus we need to trim from the end by
-        // either 2 or 3 each time.
-        if (count($path) == 6) {
-            $path = array_slice($path, -3, 2);
-        } else {
-            $path = array_slice($path, -2);
-        }
+        $icon = (object) ["icon" => null, "base" => null];
+
+        if ($observation->icon != null && strlen($observation->icon) > 0) {
+            $url = parse_url($observation->icon);
+            $path = $url["path"];
+            $path = explode("/", $path);
+
+            // An icon url, when split to path parts,
+            // with have either 5 or 6 parts.
+            // Thus we need to trim from the end by
+            // either 2 or 3 each time.
+            if (count($path) == 6) {
+                $path = array_slice($path, -3, 2);
+            } else {
+                $path = array_slice($path, -2);
+            }
 
-        $path = array_map(function ($piece) {
-            return preg_replace("/,.*$/", "", $piece);
-        }, $path);
+            $path = array_map(function ($piece) {
+                return preg_replace("/,.*$/", "", $piece);
+            }, $path);
 
-        $apiConditionKey = implode("/", $path);
+            $key = implode("/", $path);
+            $icon->icon = $this->legacyMapping->$key->icon;
```

Comment:
I guess in theory this is where an uncaught error could emerge (if there is no corresponding key on the mapping for some inexplicable reason). Kind of a note to myself for #837